src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -27155,7 +27183,7 @@ public void AllowNull_Field()
 @"using System.Diagnostics.CodeAnalysis;
 public class COpen<TOpen>
 {
-    [AllowNull]public TOpen f1 = default;
+    [AllowNull]public TOpen f1 = default!;
```

Comment:
It feels like the '!' shouldn't be necessary here. What do you think?

Reply:
It is necessary because `TOpen` could be `string!`. Then we'd be assigning `null` to a non-nullable field.

---
In reply to: [314092230](https://github.com/dotnet/roslyn/pull/37596#discussion_r314092230) [](ancestors = 314092230)